.circleci/config.yml

```diff
@@ -75,106 +66,327 @@ commands:
-      - run:
-          name: <<parameters.study_key>> deploy to GAE
-          command: gcloud app deploy --version="${CIRCLE_SHA1}" --project "broad-ddp-${ENVIRONMENT}" "${ANGULAR_PROJECT_DIR_PATH}/app.yaml"
+            set -u
+            DATE=`date +%F`
+            TAR_NAME="${ANGULAR_PROJECT_NAME}_${DATE}_${SHORT_GIT_SHA}.tar.gz"
+            TAR_PATH="/tmp/${TAR_NAME}"
+            # Save some ENV vars for in job downstream
+            echo "export TAR_NAME=$TAR_NAME" >> $BASH_ENV
+            echo "export TAR_PATH=$TAR_PATH" >> $BASH_ENV
+            cd ${OUTPUT_DIR}
+            tar -czvf ${TAR_PATH} .
+            echo "Tar file created at ${TAR_PATH}"
+            ls -l $TAR_PATH
           working_directory: ~/repo/ddp-workspace
+      - run:
+          name: Store build archive
+          command: |
+            set -u
+            ./readvault.sh secret/pepper/dev/v1/conf .data.gcp.serviceKey | gcloud auth activate-service-account --key-file=-
+            gsutil cp  ${TAR_PATH} "gs://${BUILDS_BUCKET}/${ANGULAR_PROJECT_NAME}/${TAR_NAME}"
+          working_directory: ~/repo
+
+  conditionally-launch-build-and-store:
+    description: Launch build and save job if build missing
+    parameters:
+      study_key:
+        type: string
+    steps:
+      - setup-shared-env:
+          study_key: << parameters.study_key >>
+      - run:
+          name: Initiate build for <<parameters.study_key>> and git SHA << pipeline.git.revision >> if existing one not found
+          command: |
+            set -u
+            set +e
+            # Looking for latest build of this project in Google bucket
+            ./readvault.sh secret/pepper/dev/v1/conf .data.gcp.serviceKey | gcloud auth activate-service-account --key-file=-
+            TAR_FILE_URL_PATTERN="gs://${BUILDS_BUCKET}/${ANGULAR_PROJECT_NAME}/${ANGULAR_PROJECT_NAME}_*_${SHORT_GIT_SHA}.tar.gz"
+            echo "Checking for ${TAR_FILE_URL_PATTERN}"
+            # For some reason following line generating exit code 1 every time in CircleCI. Use set +e so script can continue
+            TAR_FILE_URL=`gsutil ls $TAR_FILE_URL_PATTERN  | head -1`
 
+            # if not found, let's kick off a build
+            if [ -z $TAR_FILE_URL ]
+              then
+                echo "No build found for <<parameters.study_key>> with git sha ${SHORT_GIT_SHA}. Initiating a build for it."
+                curl -u "${CIRCLE_API_TOKEN}:" -X POST --header "Content-Type: application/json" -d '{
+                  "branch": "<< pipeline.git.branch >>",
+                  "parameters": {
+                      "study_key": "<<parameters.study_key>>",
+                      "do-builds": true
+                  }
+                }' https://circleci.com/api/v2/project/gh/broadinstitute/ddp-angular/pipeline
+              else
+                echo "Build for <<parameters.study_key>> found at URL: ${TAR_FILE_URL}. No new build will be initiated"
+            fi
```

Comment:
CIRCLE_API_TOKEN injected from the CircleCi configuration. Currently using my personal one (as far as I know there isn't a "service account" type of token).

Reply:
A bit of a disappointment w CircleCi that I needed to call the API to conditionally run individual jobs. Works well, but would be nicer if there was runtime conditionals available at the workflow level.